src/ValidationError.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Schemer;
+
+class ValidationError
+{
+    private /* const */ $messages = [
```

Comment:
Totally not averse to this, and I'm happy to yearn for the standardisation of `SplEnum`. This list will presumably have to contain all types of all errors?